### PR TITLE
Add a pre-commit hook and github action for Darker

### PR DESCRIPTION
**Pull request by theref**
**Type of PR:**
- [x] Other

**Required reviews:** 
- [X] 3

**What this does:**
Add a pre-commit hook and github action for [Darker](https://github.com/akaihola/darker) to ensure all future changes conform to black and isort.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -52,6 +52,14 @@ repos:
         files: \.py$  # only target python files, for example - skip contract_registry.json
         args: [--license-filepath, '.nucypher-license', --detect-license-in-X-top-lines=16, --comment-style, '"""||"""']
 
+  - repo: https://github.com/akaihola/darker
+    rev: 1.4.2
+    hooks:
+    -   id: darker
+        args: [--isort]
```
Comment (derekpierre) [inline .pre-commit-config.yaml:59]: Any thoughts on configuring line length - the default is `88`?

Comment (theref) [reply .pre-commit-config.yaml:59]: I've never had a problem with 88 - i know monitors have got a lot bigger, but i regularly find myself working on laptops/opening multiple files side by side/looking at diffs on github.

Whatever we choose, someone is going to be unhappy! 😁

Comment (derekpierre) [reply .pre-commit-config.yaml:59]: Haha fair enough 😄 .

**Review by derekpierre — APPROVED** [review]
🎸

**Review by cygnusv — APPROVED** [review]
LGTM. 
Re: line length, I usually have two visual rulers (80 and 120). 99% of lines I try to keep under 80, but sometimes it doesn't feel right. 120 is a hard stop for me in any case. Let's try with 88, we can always increase it a bit if we feel uncomfortable

**Review by KPrasch — APPROVED** [review]
Here we go 😅

**Pull request merged**